src/App.jsx

```diff
@@ -14,33 +16,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_ALPHABET = 'alphabet';
+const SORT_LENGHT = 'lenght';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const preparedGoods = (sortMethod, isReverse) => {
+  const gooods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  gooods.sort((a, b) => {
+    switch (sortMethod) {
+      case SORT_ALPHABET:
+        return a.localeCompare(b);
+      case SORT_LENGHT:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (isReverse) {
+    gooods.reverse();
+  }
+
+  return gooods;
+};
+
+export const App = () => {
+  const [isGoodSort, setGoodSort] = useState('');
+  const [isSortReverse, setSortIsReverse] = useState(false);
+
+  const reset = () => {
+    setGoodSort('');
+    setSortIsReverse(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const sorts = preparedGoods(isGoodSort, isSortReverse);
```

Comment:
it is a good practive to name function using verb

```suggestion
  const preparedGoods = getPreparedGoods(isGoodSort, isSortReverse);
```